src/base/LogHandler.cpp

```diff
@@ -70,4 +80,33 @@ void LogHandler::setupStdoutLogger() {
   stdoutConf.setGlobally(el::ConfigurationType::ToFile, "false");
   el::Loggers::reconfigureLogger(stdoutLogger, stdoutConf);
 }
+
+string LogHandler::createLogFile(const string &path, const string &filename) {
+  string fullFname = path + "/" + filename;
+  try {
+    fs::create_directories(path);
+  } catch (const fs::filesystem_error &fse) {
+    CLOG(ERROR, "stdout") << "Cannot create logfile directory: " << fse.what()
+                          << endl;
+    exit(1);
+  }
+#ifdef WIN32
+  // O_NOFOLLOW does not exist on windows
+  FATAL_FAIL(::open(fullFname.c_str(), O_EXCL | O_CREAT, 0600));
+#else
+  FATAL_FAIL(::open(fullFname.c_str(), O_NOFOLLOW | O_EXCL | O_CREAT, 0600));
```

Comment:
This formulation produces on failure
```
2024-11-18 11:53:08,053 FATAL [default] Stack Trace: 
[0] 0x000000010088bcab et::LogHandler::createLogFile(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)
[1] 0x000000010088b66d et::LogHandler::setupLogFiles(el::Configurations*, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, bool, bool, bool, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>)
[2] 0x00000001007d352b main
[3] 0x0000000200a562cd start
Error: (2): No such file or directory
```
can we at least log the filename? it does not seem appropriate to use FATAL_FAIL in anything other than tests (and even then it would produce hard-to-action failures).